src/sql/workbench/contrib/notebook/test/browser/notebookService.test.ts

```diff
@@ -85,4 +186,390 @@ suite('Notebook Service Tests', function (): void {
 		assert.equal(notebookService.getStandardKernelsForProvider('otherProvider').length, 1, 'otherProvider kernel info could not be found');
 		assert.deepEqual(notebookService.getStandardKernelsForProvider('otherProvider')[0], otherProviderRegistration.standardKernels, 'otherProviderRegistration standard kernels does not match');
 	});
+
+	test('tests that dispose() method calls dispose on underlying disposable objects exactly once', async () => {
+		await notebookService.registrationComplete;
+		notebookService.dispose();
+		assert.strictEqual(notebookService['_store']['_isDisposed'], true, `underlying disposable store object should be disposed state`);
+	});
+
+	test('Debug_verify that getProviderInstance does not throw when extensionService.whenInstalledExtensionRegistered() throws', async () => {
+		const providerId = 'providerId1';
+		createRegisteredProviderWithManager({ notebookService, providerId });
+		notebookService.registerProvider(providerId, undefined);
+		//verify method under test logs error and does not throw when extensionService.whenInstalledExtensionRegistered() throws
+		const error: Error = new Error('Extension Registration Failed');
+		extensionServiceMock.setup(x => x.whenInstalledExtensionsRegistered()).throws(error);
+
+		logServiceMock.setup(x => x.error(TypeMoq.It.isAny()))
+			.returns((_error: string | Error, ...args: any[]) => {
+				assert.ok(_error instanceof Error, `error passed into logService.error call should be an instance of ${Error}`);
```

Comment:
Just the strictEquals check here is enough - we already know that's an Error

Reply:
the error from the call signature could have been a string so this ensures that an instance of Error was thrown right?